src/realm/object-store/impl/object_notifier.cpp

```diff
@@ -96,13 +96,13 @@ void ObjectNotifier::run()
 
     const auto& change = it->second;
 
-    auto column_modifications = change.get_columns_modified(m_obj_key.value);
+    auto column_modifications = change.get_columns_modified(m_obj_key);
     if (!column_modifications)
         return;
 
     // Finally we add all changes to `m_change` which is later used to notify about the changed columns.
     m_change.modifications.add(0);
     for (auto col : *column_modifications) {
-        m_change.columns[col].add(0);
+        m_change.columns[col.value].add(0);
```

Comment:
how about also changing CollectionChangeSet::columns to use ColKey?

Reply:
Rabbit hole. Gave up.